module/x/gravity/keeper/grpc_query.go

```diff
@@ -307,6 +308,51 @@ func (k Keeper) ERC20ToDenom(c context.Context, req *types.ERC20ToDenomRequest)
 	return res, nil
 }
 
+func (k Keeper) DenomToERC20Params(c context.Context, req *types.DenomToERC20ParamsRequest) (*types.DenomToERC20ParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	metadata := k.bankKeeper.GetDenomMetaData(ctx, req.Denom) // lookup by base
+
+	if metadata.Base == "" { // not found, try to lookup by display
+		k.bankKeeper.IterateAllDenomMetaData(ctx, func(md banktypes.Metadata) bool {
+			if md.Display == req.Denom {
+				metadata = md
+				return true
+			}
+			return false
+		})
+	}
+
+	if metadata.Base != "" { // we have metadata, use it:
+		var (
+			erc20Name     = metadata.Display
+			erc20Symbol   = metadata.Display
+			erc20Decimals uint64
+		)
+		for _, denomUnit := range metadata.DenomUnits {
+			if denomUnit.Denom == metadata.Display {
+				erc20Decimals = uint64(denomUnit.Exponent)
+				break
+			}
+		}
+		res := &types.DenomToERC20ParamsResponse{
+			Erc20Name:     erc20Name,
+			Erc20Symbol:   erc20Symbol,
+			Erc20Decimals: erc20Decimals,
+		}
+		return res, nil
+	}
+
+	// we don't have metadata; play nice with the rules in EthereumEventProcessor.verifyERC20DeployedEvent
+	// TODO(levi) find out how we know that req.Denom is something the chain will actually regcognize:
```

Comment:
Echoing `req.Denom` without ensuring it's a token we can handle feels wrong.  We know this denom doesn't have metadata -- is there another way I can look it up?

Reply:
@jackzampolin curious to hear what your thoughts are on this

Reply:
I don't think so -- this seems correct and seems to reflect the code I recently wrote in `verifyERC20DeployedEvent`.

There is no other way to lookup metadata. Tokens without metadata will primarily be IBC tokens and we must support those. So I think returning `req.Denom` and `0` decimals is correct!

Reply:
This does seem a little weird, but it makes sense. In the absence of metadata, the only thing we care about is that the erc20 be identified with the denom. If somebody deploys an erc20 with a garbage denom, it will simply never be used.